src/lib/logical_query_plan/lqp_translator.cpp

```diff
@@ -94,9 +94,10 @@ std::shared_ptr<AbstractOperator> LQPTranslator::translate_node(const std::share
    * would result in multiple operators created from predicate_c and thus in performance drops
    */
 
-  const auto operator_iter = _operator_by_lqp_node.find(node);
-  if (operator_iter != _operator_by_lqp_node.end()) {
-    return operator_iter->second;
+  for (const auto& [l_node, p_node] : _operator_by_lqp_node) {
```

Comment:
Calling "deep" equals for an LQP Node is probably quite expensive. I'd add a hash on it that only includes the node type and a hash of its node expressions. That can be done in the abstract LQP node in <10 lines and should enable us to use a hashmap here.